.github/actions/build-prqlc/action.yaml

```
@@ -0,0 +1,54 @@
+name: build-prqlc
+description: >
+  Build prqlc
+inputs:
+  target:
+    description: Build target
+    required: true
+
+runs:
+  using: composite
+  steps:
+    - run: rustup target add ${{ inputs.target }}
+      shell: bash
+
+    - uses: Swatinem/rust-cache@v2
+      with:
+        save-if: ${{ github.ref == 'refs/heads/main' }}
+        prefix-key: 0.8.1
+
+    - if: runner.os == 'Linux'
+      shell: bash
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y musl-tools
+
+    - if: inputs.target == 'aarch64-unknown-linux-musl'
+      shell: bash
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y gcc-aarch64-linux-gnu
+        echo 'CARGO_TARGET_AARCH64_UNKNOWN_LINUX_MUSL_LINKER=aarch64-linux-gnu-gcc' >>"$GITHUB_ENV"
+        echo 'RUSTFLAGS=-C link-arg=-lgcc' >>"$GITHUB_ENV"
+
+    - name: cargo build
+      uses: richb-hanover/cargo@v1.1.0
+      with:
+        command: build
+        args: --package prqlc --release --locked --target=${{ inputs.target }}
+
+    - if: runner.os == 'Windows'
+      shell: powershell
+      run: echo "BIN_NAME=prqlc.exe" >>"$env:GITHUB_ENV"
+
+    - if: runner.os != 'Windows'
+      shell: bash
+      run: |
+        echo "BIN_NAME=prqlc" >>"$GITHUB_ENV"
+        chmod +x target/${{ inputs.target }}/release/prqlc
+
+    - name: Upload prqlc
+      uses: actions/upload-artifact@v3
+      with:
+        name: prqlc-${{ inputs.target }}
+        path: target/${{ inputs.target }}/release/${{ env.BIN_NAME }}
```

```suggestion
        path: target/${{ inputs.target }}/release/${{ runner.os == 'Windows' && 'prqlc.exe' || 'prqlc' }}
```

...is maybe less explicit but more concise, will mean we can remove most of the previous two steps (your call on which is overall better...)